### DISCUSSION TITLE
Hostname used for Verdaccio server

### DISCUSSION THREAD

**Discussion by deg** [verdaccio/verdaccio#2621] [Q&A] [answered]
I suspect this may be more of a yarn problem than of Verdaccio, but maybe you can help suggest the correct workflow.

Can I control the hostname that get placed in yarn.lock?

If I run `yarn install` on the same machine that is running my Verdaccio server, than `http://localhost:4873/...` is embedded in the generated yarn.lock.  Then, of course, if I commit this yarn.lock to my repo, it fails on any other developer's machine.

If there's a general solution for that, then I should be in great shape.

If not, please read on.  My real case is more complex, but perhaps has a more nuanced solution.

I'm developing two projects: "client" and "library". 
Library lives in Verdaccio.
Client is deployed via docker-compose and runs in a docker container.
Client's Dockerfile contains:
```
...
COPY package.json yarn.lock .yarnrc ./
RUN yarn install
COPY . ./
...
yarn start
```

This all works fine, normally.

But, sometimes I need to make changes to Library and test them in Client, before  I publish a new version of Library.
I can't (I think) use solutions like npm-link, because symlinks outside the project are not visible in the Docker container.

So, I spin up a local Verdaccio server and modify Client's `.yarnrc` to point to it.  (`registry "http://deg-basis:4873"`)

It fails with 
```
 => [4/7] RUN yarn install                                                                                                                                                                    29.9s
 => => # yarn install v1.22.5
 => => # [1/4] Resolving packages...
 => => # [2/4] Fetching packages...
 => => # error An unexpected error occurred: "http://localhost:4873/<my_library>/-/horcrux-0.0.33-dev.tgz: connect ECONNREFUSED 127.0.0.1:4873".
```

I don't even understand why localhost is appearing, since this is running in the docker container.

**Comment by juanpicado** [comment, marked as answer]
https://dev.to/verdaccio/verdaccio-and-deterministic-lock-files-19go

This is a topic out of Verdaccio context. In other words, use other package manager , pnpm or yarn2.

**Reply by deg** [reply]
Nice article!  Thanks!!

**Reply by juanpicado** [reply]
Thanks, Keep on mind that is 2 years old, situation now is different.

**Comment by deg** [comment]
(if you can spare another minute...)
What have been the relevant changes these past two years?  I guess mostly Yarn 2?  Anything else?

**Reply by juanpicado** [reply]
pnpm solved the `resolved` issue, you can see the lock file on this repo, we use pnpm 6. npm 8 still relies on the resolved field.

**Comment by juanpicado** [comment]
@deg FYI https://github.com/npm/rfcs/pull/486
